torch/csrc/jit/api/object.cpp

```diff
@@ -35,6 +35,18 @@ void Object::define(const std::string& src, const ResolverPtr& resolver) {
       *type()->name(), src, resolver ? resolver : nativeResolver(), &self);
 }
 
+Object Object::copy() const {
+  Object obj(_ivalue()->compilation_unit(), type());
+
+  size_t N = type()->numAttributes();
+  for (size_t i = 0; i < N; ++i) {
+    IValue s = _ivalue()->getSlot(i);
+    obj._ivalue()->setAttr(type()->getAttributeName(i), s.copy());
```

Comment:
calling `s.copy()` makes it a deep copy

Reply:
I see, so we just need to copy the top level wrapper for shallow copy?